src/widgets/colorpickerwidget.cpp

```diff
@@ -0,0 +1,139 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// SPDX-FileCopyrightText: 2017-2019 Alejandro Sirgo Rica & Contributors
+
+#include "colorpickerwidget.h"
+#include "src/utils/confighandler.h"
+#include "src/utils/globalvalues.h"
+#include <QMouseEvent>
+#include <QPainter>
+
+ColorPickerWidget::ColorPickerWidget(QWidget* parent)
+  : QWidget(parent)
+  , m_selectedIndex(1)
+  , m_lastIndex(1)
+{
+    initColorPicker();
+}
+
+const QVector<QColor>& ColorPickerWidget::getDefaultColors()
+{
+    return defaultColors;
+}
+
+void ColorPickerWidget::paintEvent(QPaintEvent* e)
+{
+    QPainter painter(this);
+    painter.setRenderHint(QPainter::Antialiasing);
+    painter.setPen(QColor(Qt::black));
+
+    for (int i = 0; i < m_colorAreaList.size(); ++i) {
+        if (e->region().contains(m_colorAreaList.at(i))) {
+            painter.setClipRegion(e->region());
+            repaint(i, painter);
+        }
+    }
+}
+
+void ColorPickerWidget::repaint(int i, QPainter& painter)
+{
+    // draw the highlight when we have to draw the selected color
+    if (i == m_selectedIndex) {
+        QColor c = QColor(m_uiColor);
+        c.setAlpha(155);
+        painter.setBrush(c);
+        c.setAlpha(100);
+        painter.setPen(c);
+        QRect highlight = m_colorAreaList.at(i);
+        highlight.moveTo(highlight.x() - 3, highlight.y() - 3);
```

Comment:
Why is this "-3"? Maybe 3 as a const variable with a descriptive name so its easier to read in the future?